server/src/database/archive.rs

```diff
@@ -1,92 +1,47 @@
-    start_time: i64,
-    end_time: i64,
-) -> Result<Vec<FetchTransactionResult>> {
-    let connection = &mut conn_manager
-        .archive
-        .get()
-        .context("failed to get archive db connection")?;
-
-    let results = sql_query(
-        "SELECT DISTINCT pk.value as account, uc.memo as memo, uc.nonce as nonce, uc.hash as hash, b.height as height, b.chain_status as status, b.timestamp as timestamp
-        FROM user_commands AS uc
-        JOIN blocks_user_commands AS buc
-        ON uc.id = buc.user_command_id
-        JOIN blocks AS b
-        ON buc.block_id = b.id
-        JOIN public_keys AS pk
-        ON uc.source_id = pk.id
-        WHERE uc.type = 'payment'
-        AND uc.source_id = uc.receiver_id
-        AND uc.token = 1
-        AND NOT b.chain_status = 'orphaned'
-        AND buc.status = 'applied'
-        AND b.timestamp BETWEEN $1 AND $2"
-        );
-    let results = results
-        .bind::<BigInt, _>(start_time)
-        .bind::<BigInt, _>(end_time)
-        .get_results(connection)?;
-    Ok(results)
+use std::collections::HashMap;
+
+use graphql_client::{reqwest::post_graphql_blocking as post_graphql, GraphQLQuery, Response};
+use reqwest::blocking::Client;
+
+use transaction_query::TransactionQueryTransactions;
+
+type DateTime = String;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "src/database/transaction_schema.graphql",
+    query_path = "src/database/transaction_query.graphql",
+    response_derives = "Debug"
+)]
+pub struct TransactionQuery;
+
+#[allow(clippy::unwrap_used, clippy::upper_case_acronyms)]
+pub(crate) fn fetch_transactions_graphql() {
+    let variables = transaction_query::Variables {
+        date_time_gte: Some("2023-05-20T06:00:00Z".to_string()),
+        date_time_lte: Some("2023-05-28T06:00:00Z".to_string()),
+        memo1: Some("E4YVPwLUR2LrP9tSSi3fjw1svcZys1gJHrGvRefwVTCMbP2NQRqdW".to_string()),
+        memo2: Some("E4YVe5wRALCJJ2dGEwRMyH7Z8y1QxzM76M8rXivFo5XbeBJdKryV6".to_string()),
+        memo3: Some("E4YbUmaZjNgLgezBD3JzyGKuCn4iugZ5EcXT1JuNTudm5tT4MHvKz".to_string()),
+        memo4: Some("E4YbUmaZZqAoUdTZYvZkSmLjHfccTMbb5RnTQHixwRWq2YqLdLZyE".to_string()),
+    };
+    let client = Client::new();
+    let response_body: Response<transaction_query::ResponseData> =
+        post_graphql::<TransactionQuery, _>(&client, "https://graphql.minaexplorer.com", variables)
+            .unwrap();
+
+    let txns = response_body.data.unwrap().transactions;
+
+    let txns = txns
+        .into_iter()
+        .flatten()
+        .filter(|tx| tx.to.clone().unwrap() == tx.from.clone().unwrap())
+        .fold(HashMap::new(), |mut map, txn| {
+            // Dedup based on from public key keeping the most recent transaction
+            map.entry(txn.from.clone().unwrap()).or_insert(txn);
+            map
+        })
+        .into_values()
+        .collect::<Vec<TransactionQueryTransactions>>();
+    println!("{:?}", txns.len());
```

Comment:
Translate the `TransactionQueryTransaction` into `FetchTransactionResult`. You can probably add a map after .into_values() that does the conversion and change the collect to the `Vec<FetchTransactionResult>`

Reply:
addressed by this new commit: https://github.com/Granola-Team/mina-on-chain-voting/pull/237/commits/f0eee9c4027320e62542bfc93e3caf53a619af9d